dwh/indicator_lib.py

```diff
@@ -120,4 +120,221 @@ def calc_TX_NEW(patient_agg_obs:  pd.DataFrame, ARV_plan: str,
   # Note the above copy is used to avoid setting a new column on a slice next:
   temp_df['TX_NEW'] = (temp_df['last_value_code'].isin(start_drug))
   temp_df = _gen_counts_and_ratio(temp_df, end_date, 'TX_NEW')
+  return temp_df
+
+def calc_TX_CURR(patient_agg_obs:  pd.DataFrame, ARV_plan: str,
+                ARV_plan_answer: List[str], end_date_str: str = None) -> pd.DataFrame:
+  """Calculates TX_CURR indicator with its corresponding disaggregations.
+
+  TX_CURR indicator counts the number of adults and children currently receiving
+  antiretroviral therapy (ART).
+
+  Args:
+    patient_agg_obs: A DataFrame generated by `patient_query.find_patient_aggregates()`.
+    ARV_plan: The concept question code for ANTIRETROVIRAL PLAN
+    ARV_plan_answer: The concept answer codes for START DRUG, CONTINUE REGIMEN, REFILLED
```

Comment:
AMPATH's definition for any drug/tx plan has over  5 common responses.

CONTINUE REGIMEN (1257)
START DRUGS (1256)
CHANGE REGIMEN (1259)
DOSING CHANGE (981)
CHANGE FORMULATION (1258)
REFILLED (1406)
DRUG SUBSTITUTION (1849)


Do you want us to include all these at this stage?

Reply:
I see, I suggest that as a minimum we make the comment generic, i.e., if we don't know the list of codes and it is not fixed, let's just say something like: "such as START DRUGS, etc." instead of listing them.

As some point we need to create a configuration file to make these codes easily configurable by the user but let's leave that for future.

Reply:
Agreed, also different implementers have different definitions and coding schemes